solidity/contracts/hooks/layer-zero/LayerZeroV1Hook.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+import {ILayerZeroEndpoint} from "@layerzerolabs/lz-evm-v1-0.7/contracts/interfaces/ILayerZeroEndpoint.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {MailboxClient} from "../../client/MailboxClient.sol";
+import {Indexed} from "../../libs/Indexed.sol";
+import {IPostDispatchHook} from "../../interfaces/hooks/IPostDispatchHook.sol";
+import {AbstractPostDispatchHook} from "../libs/AbstractPostDispatchHook.sol";
+import {StandardHookMetadata} from "../libs/StandardHookMetadata.sol";
+
+struct LayerZeroMetadata {
+    /// @dev the destination chain identifier
+    uint16 dstChainId;
+    /// @dev the user app address on this EVM chain. Contract address that calls Endpoint.send(). Used for LZ user app config lookup
+    address userApplication;
+    /// @dev if the source transaction is cheaper than the amount of value passed, refund the additional amount to this address
+    address refundAddress;
+    /// @dev the custom message to send over LayerZero
+    bytes payload;
+    /// @dev the address on destination chain (in bytes). A 40 length byte with remote and local addresses concatenated.
+    bytes destination;
+    /// @dev  parameters for the adapter service, e.g. send some dust native token to dstChain
+    bytes adapterParam;
+}
```

Comment:
can this also be imported? if its defined in a contract you usually do `Contract.StructName` to access

Reply:
Unfortunately not, Lz doesn't use this struct. it was created so we can encode/decode layerZeroMetadata for our use